src/dmd/mars.d

```diff
@@ -1241,6 +1241,9 @@ void addDefaultVersionIdentifiers(const ref Param params)
     {
         VersionCondition.addPredefinedGlobalIdent("Posix");
         VersionCondition.addPredefinedGlobalIdent("FreeBSD");
+        // FIXME: Need a way to statically and/or dynamically set the major FreeBSD version,
+        // to support FreeBSD 12.x and later releases both as a native and cross compiler.
+        VersionCondition.addPredefinedGlobalIdent("FreeBSD_11");
```

Comment:
Good thing Walter is working on making DMD a cross compiler. Perhaps it would be better for you to look at introducing target triplets in a way that would be compatible with GDC / LDC ?

Reply:
Note that this would also help me for the Alpine Linux package, where I want to add a specific link dependency when Musl is targeted

Reply:
LDC and most likely GDC don't use this DMD-specific `addDefaultVersionIdentifiers()`. Target triples are surely the way forward for DMD too, but I don't think there should be a standardized infrastructure here upstream - e.g., an LLVM triple is a custom type with handy methods.

Reply:
Predefined target versions aren't handled in the frontend of gdc.